### DISCUSSION TITLE
Can I add console-level padding?

### DISCUSSION THREAD

**Discussion by rlaphoenix** [Textualize/rich#2834] [Q&A]
I want my `console.log`'s, `console.print`'s, and `console.status` calls all padded 5 chars on the left and right. I have managed to get it to work for `console.log` by overriding the `_log_render` of the Console and adding padding on the `Table.grid` of the LogRenderer. For `console.status` I have overridden it with the following:

```py
status_renderable = super().status(
    status=status,
    spinner=spinner,
    spinner_style=spinner_style,
    speed=speed,
    refresh_per_second=refresh_per_second
)

if pad:
    top, right, bottom, left = Padding.unpack(pad)

    renderable_width = len(status_renderable.status)
    spinner_width = len(status_renderable.renderable.text)
    status_width = spinner_width + renderable_width

    available_width = self.width - status_width
    if available_width > right:
        # fill up the available width with padding to apply bg color
        right = available_width - right

    padding = Padding(
        status_renderable,
        (top, right, bottom, left)
    )

    return Live(
        padding,
        console=self,
        transient=True
    )

return status_renderable
```

However, I have not found an ample solution for `print`, as if I do a similar trick by wrapping all `objects` with `Padding()`, it seems to affect `Status`'s transient properties. It would randomly be transient and non-transient, seemingly depending on how fast it finished the status, or rather how little it refreshed. Or, it will not be transient ever, if the terminal width is not greater than or equal to the `console.width` by two times, for some reason.

To be clear on what I'm asking for, I want to be able to prepend 5 chars of blank (with `on color` ANSI codes) on the left and right of the entire console. E.g., if you do `console.print("Hello World")` with a theme that sets `none` to `red on white` it should render as `     Hello World     ` with the color red, and the background as white, for everything including the whitespace.

**Comment by patrick91** [comment]
This is what I'm doing at the moment, but it doesn't work for everything:

```python
class ConsoleWithPadding(Console):
    def print(self, *args, **kwargs):
        super().print(Padding(*args, (0, 2)), **kwargs)
```
